lib/bosh/workspace/merge_tool.rb

```diff
@@ -0,0 +1,76 @@
+module Bosh::Workspace
+  class MergeTool
+    include Bosh::Exec
+
+    attr_accessor :name, :version
+
+    def initialize(merge_tool = nil)
+      @name, @version = case merge_tool
+      when Hash
+        unless (%w(name version) & merge_tool.keys).size == 2
+          err("merge_tool options should contain name and version")
+        end
+        [merge_tool['name'], merge_tool['version']]
+      when String
+        [merge_tool, 'current']
+      else
+        ['spiff', 'current']
+      end
+
+      unless available_tool_names.include?(@name)
+        say("#{@name} is not supported, please specify spiff or spruce instead.".make_red)
+      end
+    end
+
+    def available_tool_names
+      %w(spiff spruce)
+    end
+
+    def merge(templates, target_file)
+      check_tool_version if version != 'current'
+      run_merge_tool(:merge, templates) do |output|
+        File.open(target_file, 'w') { |file| file.write(output) }
+      end
+    end
+
+    private
+
+    def run_merge_tool(verb, params = [])
+      params.map!(&:shellescape)
+      cmd = [name, verb.to_s] + params + ['2>&1']
+      sh(cmd.join(" "), :yield => :on_false) do |result|
+        command_not_found if result.not_found?
+        command_failed(result.command, result.output) if result.failed?
+        yield result.output
+      end
+    end
+
+    def check_tool_version
+      run_merge_tool('-v') do |output|
+        actual_version = output.match(/(\d+\.\d+\.\d+)/).to_a.first
+        if actual_version.nil? || actual_version != version
+          say "[WARNING] Deployment requires #{name} to have version #{version}\n".make_yellow
```

Comment:
If I'm not mistaken you can use the [`warning`](https://github.com/cloudfoundry/bosh/blob/master/bosh_cli/lib/cli/core_ext.rb#L111) method here.